### ISSUE TITLE
Feature request : await()/await(duration) methods

### ISSUE THREAD

**Issue by jvmlet** [ctongfei/progressbar#96]
Would you  please add await capability to the progress bar : 
```java
c'tor :
latch = CountDownLatch(initialMax)

on stepBy :
 latch.countDown(n times)

and
public void  await(){
latch .await()
}
publiv void  await(duration){
latch .await(duration)
}
```

**Comment by ctongfei** [comment]
Hi @jvmlet !
I don't fully understand what you mean here: what is the semantics of `await`?

**Comment by jvmlet** [comment]
Blocks until latch reaches 0 (  100% progress).

**Comment by ctongfei** [comment]
Blocking the main thread until the progress finishes?

**Comment by jvmlet** [comment]
Yes

**Comment by ctongfei** [comment]
Should this be a progress bar's job? I suppose that when a user starts a task that is tracked by a progress bar, one can always await on that task instead of on the progress bar.

**Comment by jvmlet** [comment]
You are right, but I most cases completion is tracked by count down latch. I thought that if progress bar has this count, it would be easier to this via progress bar

**Comment by ctongfei** [comment]
In the spirit of decoupling and modularizing software modules, I have decided not to implement this feature since this is out of scope of a progress bar. Thanks @jvmlet for the suggestion.
